lib/adapters/class-adapter.php

```diff
@@ -303,6 +310,111 @@ public function add_term_aggregation( string $label, string $term_field, string
 		);
 	}
 
+	/**
+	 * A callback to filter es args.
+	 * Adds phrase matching to the request args if it is enabled.
+	 *
+	 * @param array $es_args The request args to be filtered.
+	 * @return array The filtered request args.
+	 */
+	public function add_phrase_matching_to_es_args( array $es_args ): array {
+		if ( ! $this->get_enable_phrase_matching() ) {
+			return $es_args;
+		}
+
+		$search = get_query_var( 's' );
+
+		// Bail early if this isn't a search.
+		if ( empty( $search ) ) {
+			return $es_args;
+		}
+
+		// Break down the search string into the desired "phrase matched" parts.
+		// TODO: Add filter for phrase_match_delineator.
+		$phrase_match_delineator = '"';
+		preg_match_all( '/' . $phrase_match_delineator . '(.*?)' . $phrase_match_delineator . '/', $search, $matches );
+		if ( empty( $matches ) ) {
```

Comment:
Is it worth checking for `$matches[1]` instead of `$matches`?

Reply:
if `$matches` isn't empty then `$matches[1]` is implied (since there has to be at least one match for `$matches` to be defined). At any rate, it is good to be explicit, so I will update it.